### PR TITLE
dashboard: control dashboard running node

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
We start dashboard services in all PDs now and actually we only need one. It can reduce the waste of resources.

### What is changed and how it works?
This PR adds a mechanism to control the dashboard running node by using `dashbaord-address` parameters. It has three kinds of values:
- `auto`: select the sever whose member ID is the minimum to run dashboard.
- `none`: stop all dashboard services
- specified address like `127.0.0.1:2379`: select the given address to run dashboard.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: pkg/dashboard/manager.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+		m.stopService()
+	}
+
+	return known
+}
+
+func (m *Manager) isOnline(addr string) bool {
+	rc := m.srv.GetRaftCluster()
+	if rc == nil || !rc.IsRunning() {
+		return true
```
Comment (breeswish) [inline pkg/dashboard/manager.go]: why it is online in this case?

File: pkg/dashboard/manager.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+	}
+	members, err := cluster.GetMembers(m.srv.GetClient())
+	if err != nil || len(members) == 0 {
+		log.Error("failed to get members")
+		return
+	}
+
+	// get new dashboard address
+	var addr string
+	switch len(members) {
```
Comment (breeswish) [inline pkg/dashboard/manager.go]: in what case there will be 0 members?

File: pkg/dashboard/manager.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+	case 0:
+		return
+	case 1:
+		addr = members[0].GetClientUrls()[0]
+	default:
+		addr = members[0].GetClientUrls()[0]
+		leaderID := m.srv.GetMemberInfo().MemberId
+		sort.Slice(members, func(i, j int) bool { return members[i].GetMemberId() < members[j].GetMemberId() })
```
Comment (breeswish) [inline pkg/dashboard/manager.go]: why do we need to sort the member id?

File: pkg/dashboard/manager.go [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+				addr = member.GetClientUrls()[0]
+				break
+			}
+		}
+	}
+
+	// set new dashboard address
+	rc.SetDashboardAddress(addr)
+	if m.srv.GetConfig().EnableDynamicConfig {
+		if err := m.srv.UpdateConfigManager("pd-server.dashboard-address", addr); err != nil {
```
Comment (breeswish) [inline pkg/dashboard/manager.go]: should it be a global config?

File: pkg/dashboard/servers.go [hunk 5]
```diff
@@ -44,28 +42,19 @@ var (
...
-			})
-			srv.AddCloseCallback(func() {
-				if err := s.Stop(context.Background()); err != nil {
-					log.Error("Stop dashboard server error", zap.Error(err))
-				} else {
-					log.Info("Dashboard server is stopped")
-				}
-			})
+			m := NewManager(srv, s, redirector)
+			srv.AddStartCallback(m.start)
```
Comment (breeswish) [inline pkg/dashboard/servers.go]: is it possible that start can be called multiple times? @HunDunDM

Comment (HunDunDM) [reply pkg/dashboard/servers.go]: This life cycle is consistent with PD Server. I think it cannot be called multiple times.

File: pkg/dashboard/manager.go (hunk 4 above)
Comment (HunDunDM) [reply pkg/dashboard/manager.go]: @rleungx Is it possible to change this configuration at runtime?

File: pkg/dashboard/manager.go (hunk 3 above)
Comment (HunDunDM) [reply pkg/dashboard/manager.go]: We want to select the follower node with the smallest member id.

File: pkg/dashboard/manager.go (hunk 1 above)
Comment (HunDunDM) [reply pkg/dashboard/manager.go]: At this time, it is impossible to determine whether it is online, so it cannot be handled as offline.

File: pkg/dashboard/manager.go (hunk 4 above)
Comment (HunDunDM) [reply pkg/dashboard/manager.go]: No change during runtime

File: pkg/dashboard/manager.go (hunk 2 above)
Comment (HunDunDM) [reply pkg/dashboard/manager.go]: It cannot be 0 here, I have deleted it.

File: pkg/dashboard/manager.go [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+	switch len(members) {
+	case 1:
+		addr = members[0].GetClientUrls()[0]
+	default:
+		addr = members[0].GetClientUrls()[0]
+		leaderID := m.srv.GetMemberInfo().MemberId
+		sort.Slice(members, func(i, j int) bool { return members[i].GetMemberId() < members[j].GetMemberId() })
+		for _, member := range members {
+			if member.MemberId != leaderID {
+				addr = member.GetClientUrls()[0]
```
Comment (nolouch) [inline pkg/dashboard/manager.go]: what if the join a new member or leader changed?

Comment (HunDunDM) [reply pkg/dashboard/manager.go]: Nothing happens unless the member that is the dashboardAddress is deleted.

**Pull request merged**